plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/delete/IcebergPositionDeletePageSink.java

```diff
@@ -0,0 +1,162 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.iceberg.delete;
+
+import io.airlift.json.JsonCodec;
+import io.airlift.slice.Slice;
+import io.trino.plugin.hive.HdfsEnvironment;
+import io.trino.plugin.hive.HdfsEnvironment.HdfsContext;
+import io.trino.plugin.iceberg.CommitTaskData;
+import io.trino.plugin.iceberg.IcebergFileWriter;
+import io.trino.plugin.iceberg.IcebergFileWriterFactory;
+import io.trino.plugin.iceberg.MetricsWrapper;
+import io.trino.plugin.iceberg.PartitionData;
+import io.trino.spi.Page;
+import io.trino.spi.connector.ConnectorPageSink;
+import io.trino.spi.connector.ConnectorSession;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.mapred.JobConf;
+import org.apache.iceberg.FileContent;
+import org.apache.iceberg.FileFormat;
+import org.apache.iceberg.PartitionSpec;
+import org.apache.iceberg.Schema;
+import org.apache.iceberg.io.LocationProvider;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+
+import static io.airlift.slice.Slices.wrappedBuffer;
+import static io.trino.plugin.hive.util.ConfigurationUtils.toJobConf;
+import static java.util.Objects.requireNonNull;
+import static java.util.UUID.randomUUID;
+import static java.util.concurrent.CompletableFuture.completedFuture;
+import static org.apache.iceberg.io.DeleteSchemaUtil.pathPosSchema;
+
+public class IcebergPositionDeletePageSink
+        implements ConnectorPageSink
+{
+    private static final int MAX_PAGE_POSITIONS = 4096;
+    private static final Schema POSITION_DELETE_SCHEMA = pathPosSchema();
+
+    private final Optional<PartitionData> partition;
+    private final String outputPath;
+    private final JsonCodec<CommitTaskData> jsonCodec;
+    private final IcebergFileWriter writer;
+
+    private long writtenBytes;
+    private long systemMemoryUsage;
+    private long validationCpuNanos;
+    private boolean writtenData;
+
+    public IcebergPositionDeletePageSink(
```

Comment:
My understanding is that all deletes received will always come from the same split, which means the same partition, so there is no need to use the fanout writer. Please let me know if it is not the case.